testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/realm/FilesystemRealmEncryptedTestCase.java

```diff
@@ -0,0 +1,209 @@
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.as.arquillian.api.ServerSetup;
+import org.jboss.as.arquillian.api.ServerSetupTask;
+import org.jboss.as.arquillian.container.ManagementClient;
+import org.jboss.as.test.integration.management.util.CLIWrapper;
+import org.jboss.as.test.integration.security.common.Utils;
+import org.jboss.as.test.integration.security.common.servlets.SimpleSecuredServlet;
+import org.jboss.as.test.integration.security.common.servlets.SimpleServlet;
+import org.jboss.as.test.shared.ServerReload;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * Full authentication tests for Elytron Encrypted Filesystem Realm.
+ *
+ * @author Ashpan Raskar <araskar@redhat.com>
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+@ServerSetup({FilesystemRealmEncryptedTestCase.SetUpTask.class})
+public class FilesystemRealmEncryptedTestCase {
+
+    private static final String DEPLOYMENT_ENCRYPTED = "filesystemRealmEncrypted";
+    private static final String USER = "plainUser";
+    private static final String PASSWORD = "secretPassword";
+
+
+    @Deployment(name = DEPLOYMENT_ENCRYPTED)
+    public static WebArchive deploymentWithCharsetAndEncoded() {
+        final WebArchive war = ShrinkWrap.create(WebArchive.class, DEPLOYMENT_ENCRYPTED + ".war");
+        war.addClasses(SimpleServlet.class);
+        war.addClasses(SimpleSecuredServlet.class);
+        war.addAsWebInfResource(FilesystemRealmEncryptedTestCase.class.getPackage(), "filesystem-realm-web.xml", "web.xml");
+        war.addAsWebInfResource(Utils.getJBossWebXmlAsset(DEPLOYMENT_ENCRYPTED), "jboss-web.xml");
+        return war;
+    }
+
+    /**
+     *
+     * Test Filesystem realm when choosing to encrypt it.
+     */
+    @Test
+    @OperateOnDeployment(DEPLOYMENT_ENCRYPTED)
+    public void testEncryptionPass(@ArquillianResource URL webAppURL) throws Exception {
+        URL url = prepareURL(webAppURL);
+        Utils.makeCallWithBasicAuthn(url, USER, PASSWORD, SC_OK);
+    }
+
+    /**
+     *
+     * Test Filesystem realm when choosing to encrypt it with a bad credential.
+     */
+    @Test
+    @OperateOnDeployment(DEPLOYMENT_ENCRYPTED)
+    public void testEncryptionCredentialFail(@ArquillianResource URL webAppURL) throws Exception {
+        URL url = prepareURL(webAppURL);
+        Utils.makeCallWithBasicAuthn(url, USER+"123", PASSWORD, SC_UNAUTHORIZED);
+    }
+
+    /**
+     *
+     * Test Filesystem realm when choosing to encrypt it with a bad Secret Key.
+     */
+    @Test
+    @OperateOnDeployment(DEPLOYMENT_ENCRYPTED)
+    public void testEncryptionSecretKeyFail(@ArquillianResource URL webAppURL) throws Exception {
+        URL url = prepareURL(webAppURL);
+        SetUpTask.setUpBadCredentialStore(SetUpTask.REALM_NAME, SetUpTask.CREDENTIAL_STORE+"New");
```

Comment:
Because the bad credential store has a different name from the good one, you should be able to do the setup in the regular `setup` method below. That way, you won't need a boolean to track whether or not it's been created.